src/canvas.rs

```diff
@@ -97,6 +127,46 @@ impl<P: AsBytes + FromBytes> Canvas<P> {
         Self::with_layout(layout)
     }
 
+    /// Interpret an existing buffer as an image.
+    ///
+    /// The data already contained within the buffer is not modified so that prior initialization
+    /// can be performed or one array of samples reinterpreted for an image of other sample type.
+    /// However, the `Rec` will be logically resized which will zero-initialize missing elements if
+    /// the current buffer is too short.
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if resizing causes a reallocation that fails.
+    pub fn from_rec(mut buffer: Rec<P>, layout: Layout<P>) -> Self {
+        buffer.resize_bytes(layout.byte_len());
+        Self::new_raw(buffer, layout)
+    }
+
+    /// Reuse an existing buffer for an image.
+    ///
+    /// Similar to `from_rec` but this function will never reallocate the inner buffer. Instead, it
+    /// will return the `Rec` unmodified in the `Err` variant.
+    pub fn reuse_rec(mut buffer: Rec<P>, layout: Layout<P>)
```

Comment:
Perhaps something like `from_rec_reuse`. Also "it will return the Rec unmodified... *if the creation fails*"